cpp/src/decisiontree/levelalgo/levelfunc_classifier.cuh

```diff
@@ -0,0 +1,136 @@
+/*
```

Comment:
It seems like still most of the code between levelfunc_classifier.cuh and level_func_regressor.cuh is the same. Can we have them share the common logic, please?

Reply:
Hmm, that would be hard, they have different data structures and a slightly different workflow. Mixing them will involve lots of if and else. Hard to understand.

Right now we have a clear distinction in the code flow. I find this more manageable

Reply:
I see. In that case, let's postpone for any such commonalities in the future PR's.